src/firebaseFunctions.tsx

```diff
@@ -0,0 +1,75 @@
+import { push, ref, update } from "firebase/database";
+import { auth, database } from "./firebase";
+
+type EventInfoDataType = {
+  eventName: string;
+  eventStartDate: string;
+  eventEndDate: string;
+};
+
+type EventDetailsDataType = {
+  eventName: string;
+  eventDescription: string;
+  eventStartDate: string;
+  eventEndDate: string;
+  eventLocation: string;
+  eventCategory: string;
+};
+
+type EventData = {
+  eventName: string;
+  eventDescription: string;
+  eventStartDate: string;
+  eventEndDate: string;
+  eventLocation: string;
+  eventCategory: string;
+};
+
+export function addEvent(eventData: EventData) {
+  const newEventKey = push(ref(database, "events")).key;
+  if (newEventKey === null) {
+    return;
+  }
+  const eventInfoData = {
+    eventName: eventData.eventName,
+    eventStartDate: eventData.eventStartDate,
+    eventEndDate: eventData.eventEndDate,
+  };
+  const eventDetailsData = {
+    eventName: eventData.eventName,
+    eventDescription: eventData.eventDescription,
+    eventOrganizer: auth.currentUser?.uid,
```

Comment:
i would pussh the event if we have a user
the "?" means that we might not have a user.